### DISCUSSION TITLE
How to detect if a location is "water" or "land" etc..

### DISCUSSION THREAD

**Discussion by shay-te** [rnmapbox/maps#1165] [Q&A]
Hello all. 
How to know if a point is on "water"(ocean) or land/city/etc.
when the user picks a point I want to know this location is valid. 

I know this is possible using the `map.visibleFeatures` in the `javascript` version. 

I created a StackOverflow question. but didn't find an answer yet. 
https://stackoverflow.com/questions/65548816/cant-find-function-visiblefeatures

Thank you for the help, 
Shay

**Comment by ferdicus** [comment]
this stack answer mentions an easy solution: https://stackoverflow.com/a/44238234/1892871

check the available methods in the docs for MapView
https://github.com/react-native-mapbox-gl/maps/blob/master/docs/MapView.md

**Reply by shay-te** [reply]
Thank you very much for your help. the function 'visibleFeatures' is not available in the MapView documentation.

**Reply by ferdicus** [reply]
yes, but there are equivalent methods to the JS apis `queryRenderedFeatures`, like:
`queryRenderedFeaturesAtPoint`
`queryRenderedFeaturesInRect`

**Reply by shay-te** [reply]
yes, already tried that and didn't mention it.  the output is not telling a lot.

```javascript
var MyMap = function(props) {
    const mapContainer = useRef(null);
...
    function onMapPointSelect(data) {
            mapContainer.current.queryRenderedFeaturesAtPoint(data.geometry.coordinates).then(function(result) {
                console.log(result);
                for(var i = 0 ; i < result.features.length ; i++) {
                    console.log(result.features[i]);
                }
            })
      }


      return (
            <MapboxGL.MapView ref={el => (mapContainer.current = el)} onPress={onMapPointSelect}  .......> 
                     .....   
             </MapboxGL.MapView> )
     ) ;  
}

```



when clicking the output is : 

```shell 
[Mon Jan 18 2021 14:44:18.350]  LOG      {"features": [{"geometry": [Object], "id": "", "properties": [Object], "type": "Feature"}], "type": "FeatureCollection"}
[Mon Jan 18 2021 14:44:18.360]  LOG      {"geometry": {"coordinates": [[Array], [Array], [Array], [Array], [Array], [Array], [Array], [Array], [Array], [Array], [Array], [Array], [Array], [Array], [Array], [Array]], "type": "MultiPolygon"}, "id": "", "properties": {}, "type": "Feature"}
```

I am probably doing something wrong. 
Thank you for the help!

**Reply by shay-te** [reply]
@ferdicus can you help, please. 
Thank you

**Reply by ferdicus** [reply]
please go through the stackoverflow link, it mentions how to use this approach

**Reply by shay-te** [reply]
this link [https://stackoverflow.com/questions/19775585/mapbox-water-land-detection/44238234#44238234](https://stackoverflow.com/questions/19775585/mapbox-water-land-detection/44238234#44238234)? 
instructs how to use the `javascript` version `visibleFeatures`.
this method does not exist in the `react-native` version.

can you be more specific, please?

**Reply by ferdicus** [reply]
My assumption was, that the methods,
`queryRenderedFeaturesAtPoint` or
`queryRenderedFeaturesInRect`

Would return something with which you could do the test as mentioned in the stackoverflow link:
> If empty result: no water, if polygon: water.

If that is not possible with the provided methods, I'm sorry, I'm out of ideas - you should turn to the chat for guidance

**Reply by shay-te** [reply]
Thank you.

**Comment by shay-te** [comment]
Any help, please?
